app/Http/Controllers/Bible/BiblesController.php

```diff
@@ -702,98 +707,106 @@ public function chapter(Request $request, $bible_id)
         $result->chapter = $chapter;
 
         if ($copyrights) {
-            $result->copyrights = $this->copyright($bible->id)->original;
+            $result->copyrights = cacheRemember('v4_chapter_copyrights', [$bible->id], now()->addDay(), function () use ($bible) {
+                return $this->copyright($bible->id)->original;
+            });
         }
 
-        $chapter_filesets = (object) [
-            'video' => (object) ['gospel_films' => [], 'jesus_films' => []],
-            'audio' => (object) [],
-            'text' => (object) [],
-            'timestamps' => (object) [],
-        ];
+        $cache_params = [$bible_id, $book_id, $chapter, $zip, $drama];
 
-        if ($zip) {
-            $chapter_filesets->downloads = [];
-        }
+        $chapter_filesets = cacheRemember('v4_chapter_filesets', $cache_params, now()->addHours(12), function () use ($drama, $zip, $bible, $book, $bible_id, $book_id, $chapter) {
```

Comment:
The filesets are 12 hours because we have: 

- 12 hours of client cache (app-side)
- 24 of the signed mp3 URLs
- 12 hours of API cache

This way the mp3 url is always valid.